website/addons/base/__init__.py

```diff
@@ -627,6 +631,27 @@ def get_attached_nodes(self, external_account):
             if node_settings.external_account == external_account:
                 yield node
 
+    def merge(self, user_settings):
+        """Merge `user_settings` into this instance"""
+        if user_settings.__class__ is not self.__class__:
+            raise TypeError('Cannot merge different addons')
+
+        for node_id, data in user_settings.oauth_grants.iteritems():
+            if node_id not in self.oauth_grants:
+                self.oauth_grants[node_id] = data
+            else:
+                for ext_acct, meta in user_settings.oauth_grants[node_id]:
```

Comment:
Yep, good catch. Fixed in ca7bbc7 - this section could use additional testing, since this wasn't caught.